lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -46,7 +46,6 @@ def matches?(given_proc, negative_expectation=false, &block)
           @eval_block = false
           @eval_block_passed = false
 
-          warn_about_bare_error if warning_about_bare_error && !negative_expectation
           return false unless Proc === given_proc
```

Comment:
I'm surprised that moving this after the `return false` didn't cause anything to fail.

That would be this case I think:

```
> rspec test_spec.rb
WARNING: Using the `raise_error` matcher without providing a specific error or message risks false positives, since `raise_error` will match when Ruby raises a `NoMethodError`, `NameError` or `ArgumentError`, potentially allowing the expectation to pass without even executing the method you are intending to call. Instead consider providing a specific error class or message. This message can be supressed by setting: `RSpec::Expectations.configuration.warn_about_potential_false_positives = false`. Called from /private/tmp/test_spec.rb:3:in `block (2 levels) in <top (required)>'.
F

Failures:

  1) blah should raise Exception
     Failure/Error: expect(nil).to raise_error
       expected Exception but was not given a block
     # ./test_spec.rb:3:in `block (2 levels) in <top (required)>'

Finished in 0.0212 seconds (files took 0.21044 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./test_spec.rb:2 # blah should raise Exception
```

I'm ok with not issuing a warning here, and don't consider warnings part of backwards compatibility guarantee.